File: docs-examples/codecs-java/src/test/java/io/micronaut/configuration/lettuce/docs/NamedConnectionCommands.java
package io.micronaut.configuration.lettuce.docs;

import io.lettuce.core.api.StatefulRedisConnection;
import io.micronaut.context.annotation.Requires;
import jakarta.inject.Inject;
import jakarta.inject.Named;
import jakarta.inject.Singleton;

@Requires(property = "spec.name", value = "NamedConnectionTest")
@Singleton
public final class NamedConnectionCommands {

    // tag::namedConnection[]
    @Inject @Named("foo") StatefulRedisConnection<String, String> connection;
    // end::namedConnection[]

    void commands() {
        connection.sync().set("named", "foo");
    }
}

File: docs-examples/codecs-kotlin/src/test/kotlin/io/micronaut/configuration/lettuce/docs/NamedConnectionCommands.kt
package io.micronaut.configuration.lettuce.docs

import io.lettuce.core.api.StatefulRedisConnection
import io.micronaut.context.annotation.Requires
import jakarta.inject.Inject
import jakarta.inject.Named
import jakarta.inject.Singleton

@Requires(property = "spec.name", value = "NamedConnectionTest")
@Singleton
class NamedConnectionCommands {

    // tag::namedConnection[]
    @Inject
    @field:Named("foo")
    lateinit var connection: StatefulRedisConnection<String, String>
    // end::namedConnection[]

    fun commands() {
        connection.sync().set("named", "foo")
    }
}
